bolt/lib/Passes/LongJmp.cpp

```diff
@@ -382,6 +382,13 @@ uint64_t LongJmpPass::tentativeLayoutRelocMode(
     DotAddress += Func->estimateConstantIslandSize();
     ++CurrentIndex;
   }
+
+  // Ensure that tentative code layout always runs for cold blocks.
+  if (!ColdLayoutDone)
+    runColdLayout();
+  assert(ColdLayoutDone &&
```

Comment:
It seems this assert is redundant and could not be raised by anything, I suggest to remove it

Reply:
True. I kept it just for testing but TBH I also did not like it.

This is now gone. I'm using a mock profile to trigger this, and I check whether any cold tentative estimation happened.

I've also updated the [description of this PR](https://github.com/llvm/llvm-project/pull/96609#issue-2372128097).